mlir/lib/Conversion/GPUToLLVMSPV/GPUToLLVMSPV.cpp

```diff
@@ -385,6 +385,19 @@ struct GPUToLLVMSPVConversionPass final
     if (failed(applyPartialConversion(getOperation(), target,
                                       std::move(patterns))))
       signalPassFailure();
+
+    // `func.func`s are not handled by the lowering, so need a proper calling
+    // convention set separately.
+    getOperation().walk([&](LLVM::LLVMFuncOp f) {
+      if (f.getCConv() == LLVM::CConv::C) {
+        f.setCConv(LLVM::CConv::SPIR_FUNC);
+      }
+    });
+    getOperation().walk([&](LLVM::CallOp c) {
```

Comment:
I don't think these two lambdas capture anything?

```suggestion
    getOperation().walk([](LLVM::LLVMFuncOp f) {
      if (f.getCConv() == LLVM::CConv::C) {
        f.setCConv(LLVM::CConv::SPIR_FUNC);
      }
    });
    getOperation().walk([](LLVM::CallOp c) {
```